CONTRIBUTING.md

```diff
@@ -1,19 +1,59 @@
-# Contributing to this registry
+# Contributing
 
 This document outlines the requirements for contributing a devfile stack or sample to this repository.
 
 The [devfile registry structure](https://github.com/devfile/api/blob/main/docs/proposals/registry/registry-structure.md#repository-structure) design doc provides some useful background information on the structure of resources in a devfile registry (and its Git repository).
 
 When onboarding a new stack or sample, the  `Stack Provider` should read and agree to follow their roles and responsibilities outlined in the [Lifecycle](LIFECYCLE.md) doc.
 
-## Prerequisites
+## Code of Conduct
+
+Before contributing to this repository for the first time, please review our project's [Code of Conduct](https://github.com/devfile/api/blob/main/CODE_OF_CONDUCT.md).
+
+## Certificate of Origin
+
+By contributing to this project you agree to the Developer Certificate of
+Origin (DCO). This document was created by the Linux Kernel community and is a
+simple statement that you, as a contributor, have the legal right to make the
+contribution. See the [DCO](DCO) file for details.
+
+In order to show your agreement with the DCO you should include at the end of the commit message,
+the following line:
+
+```console
+Signed-off-by: Firstname Lastname <email@email.com>
+```
+
+Once you set your user.name and user.email in your git config, you can sign your commit automatically with `git commit -s`.
+
+## How to Contribute:
+
+### Issues
+
+For issues relating to a specific devfile stack in this repository, please reach out to the devfile stack maintainer to determine where to open an issue.
+
+For issues relating to the hosted devfile registry service (<https://registry.devfile.io>), or devfile registries in general, please use the [devfile/api](https://github.com/devfile/api/) issue tracker for opening issues. Apply the `area/registry` label to registry issues for better visibility.
+
+### Submitting Pull Request
+
+When you think the code is ready for review, create a pull request and link the issue associated with it.
+
+Owners of the repository will watch out for new PRs and provide reviews to them.
+
+If comments have been given in a review, they have to be addressed before merging.
+
+After addressing review comments, don't forget to add a comment in the PR with the reviewer mentioned afterward, so they get notified by Github to provide a re-review.
+
+See [Registry Stack Review Process](#registry-stack-review-process) for more information about the review process of devfile registry content.
+
+### Prerequisites
 
 The following are required to build the devfile index container image containing your stack or sample:
 
-- Docker 17.06 or later
+- Docker or Podman
 - Git
 
-## Instructions
+### Instructions
 
 1. Open an issue in the [devfile/api](https://github.com/devfile/api) repo to track adding a new stack or sample
```

Comment:
I think this is also mentioned above in the `How to Contribute` > `Issues` section. Apart from that, as a reader is not clear to me what the instructions is for? (e.g is for "how I could contribute" or something else)

Reply:
@thepetk 

> I think this is also mentioned above in the `How to Contribute` > `Issues` section.

Referred back the the `Issues` section for this step.

> Apart from that, as a reader is not clear to me what the instructions is for? (e.g is for "how I could contribute" or something else)

Yeah I agree this wasn't very clear, how does this revision sound? https://github.com/devfile/registry/pull/421/commits/08bef5147547ace067d72fecd03d3a32ad41f631